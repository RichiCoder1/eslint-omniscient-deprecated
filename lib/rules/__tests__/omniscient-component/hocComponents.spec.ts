import { fix } from "./__setup__/setup";

describe("hoc components", () => {
    it("should handle wrapped components", () => {
        const input = `
import component from 'omniscient';

function withProps(BaseComponent, props) {
    return component(BaseComponent.displayName + '_WithProps', newProps => {
        return <Component {...newProps} {...props} />;
    });
}`;
        const report = fix(input);
        expect(report).toBeFixed();
        expect(report).toMatchSnapshot();
    });
});
