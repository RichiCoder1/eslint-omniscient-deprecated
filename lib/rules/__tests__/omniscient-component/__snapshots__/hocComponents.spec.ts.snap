// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hoc components should handle wrapped components 1`] = `
Object {
  "fixed": true,
  "messages": Array [
    Object {
      "column": 28,
      "fatal": true,
      "line": 21,
      "message": "Parsing error: Unexpected token =",
      "ruleId": null,
      "severity": 2,
    },
  ],
  "output": "import { Component } from 'react';
import { immutShouldComponentUpdate } from 'TestModule';

import component from 'omniscient';

function withProps(BaseComponent, props) {
    return (() => {
    class AnonymousComponent extends Component {
        shouldComponentUpdate(nextProps, nextState) {
            return (
                !immutShouldComponentUpdate(this.props, nextProps) ||
                !immutShouldComponentUpdate(this.state, nextState)
            );
        }

        render() {
            const newProps = this.props;
            return <Component {...newProps} {...props} />;
        }

        static displayName = BaseComponent.displayName + \\"_WithProps\\";
    }

    return AnonymousComponent;
})();
}",
}
`;
